lib/client/http-tracker.js

```diff
@@ -124,56 +132,59 @@ class HTTPTracker extends Tracker {
 
     this.cleanupFns.push(cleanup)
 
-    let request = get.concat({
-      url: parsedUrl.toString(),
-      agent,
-      timeout: common.REQUEST_TIMEOUT,
-      headers: {
-        'user-agent': this.client._userAgent || ''
-      }
-    }, onResponse)
-
-    function cleanup () {
-      if (request) {
-        arrayRemove(self.cleanupFns, self.cleanupFns.indexOf(cleanup))
-        request.abort()
-        request = null
-      }
-      if (self.maybeDestroyCleanup) self.maybeDestroyCleanup()
+    let res
+    let controller = abortTimeout(common.REQUEST_TIMEOUT)
+    try {
+      res = await fetch(parsedUrl.toString(), {
+        agent,
+        signal: controller.signal,
+        dispatcher: agent,
```

Comment:
the 3rd party `node-fetch` package supports the agent field since internally it uses http, however the node native fetch is built using undicti, so it doesn't support the agent field, instead it has a dispatcher field, but it's not compatible with the http agent, some translation/proxy package would be needed for this
https://undici.nodejs.org/#/docs/api/Dispatcher